core/chains/evm/logpoller/orm.go

```diff
@@ -386,23 +418,86 @@ type Exp struct {
 }
 
 func (o *DSORM) SelectUnmatchedLogIDs(ctx context.Context, limit int64) (ids []uint64, err error) {
-	query := `
-		SELECT l.id FROM evm.logs l LEFT JOIN (
+	batchLogsSubQuery := `SELECT id, evm_chain_id, address, event_sig FROM evm.logs
+                                WHERE evm_chain_id = $1 AND block_number >= $2 AND block_number <= $3`
+
+	query := fmt.Sprintf(`
+		SELECT l.id FROM (%s) l LEFT JOIN (
 			SELECT evm_chain_id, address, event
 			FROM evm.log_poller_filters
 				WHERE evm_chain_id = $1
 				GROUP BY evm_chain_id, address, event
 		) r ON l.evm_chain_id = r.evm_chain_id AND l.address = r.address AND l.event_sig = r.event
 		WHERE l.evm_chain_id = $1 AND r.evm_chain_id IS NULL
-	`
+	`, batchLogsSubQuery)
 
-	if limit == 0 {
-		err = o.ds.SelectContext(ctx, &ids, query, ubig.New(o.chainID))
+	latestBlock, err := o.SelectLatestBlock(ctx)
+	if err != nil {
 		return ids, err
 	}
-	err = o.ds.SelectContext(ctx, &ids, fmt.Sprintf("%s LIMIT %d", query, limit), ubig.New(o.chainID))
 
-	return ids, err
+	r := NewRangeQueryer[uint64](o.chainID, o.ds, func(ctx context.Context, r *RangeQueryer[uint64], lower, upper int64) (int64, error) {
+		var rowIDs []uint64
+		err2 := r.ds.SelectContext(ctx, &rowIDs, query, r.chainID, lower, upper)
+		if err2 != nil {
+			return 0, err2
+		}
+		r.AddResults(rowIDs)
```

Comment:
I wonder if returning `rowIDs` from `query` function wouldn't be cleaner in terms of understanding the execution flow. I don't have strong opinion on that. 

I mean this
```go
		err2 := r.ds.SelectContext(ctx, &rowIDs, query, r.chainID, lower, upper)
		if err2 != nil {
			return 0, err2
		}
		return rowIDs, int64(len(rowIDs)), nil
```

Reply:
I would agree if they were all SELECT queries, but I don't like having to return an extra empty list for DELETE, INSERT, or UPDATE queries, since the # of rows affected is the only actual result in that case. Requiring the SELECT queries to call a separate method makes more sense to me, since it's an optional feature not needed for most types of queries (although we're currently only using it for DELETE and SELECT).